src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -83,6 +83,9 @@ private static async ValueTask<(bool, T?)> TryResolveBodyAsync<T>(HttpContext ht
 """;
 
     public static string WriteToResponseAsyncMethod => """
+        [UnconditionalSuppressMessage("Trimming", "IL2026:RequiresUnreferencedCode",
+            Justification = "<Pending>")]
+        [UnconditionalSuppressMessage("AOT", "IL3050:RequiresDynamicCode", Justification = "See above.")]
         private static Task WriteToResponseAsync<T>(T? value, HttpContext httpContext, JsonTypeInfo<T> jsonTypeInfo, JsonSerializerOptions options)
```

Comment:
Having the method accept both a `JsonTypeInfo<T>` and a `JsonSerializerOptions` parameter seems a bit redundant. Assuming it's meant to accept a pre-cached `JsonTypeInfo` instance then it might make sense to an assert checking that
```C#
Debug.Assert(jsonTypeInfo.Options == options);
```
but otherwise it looks like either of the two parameters should be removed.

Reply:
Yes, we pre-compute the `JsonTypeInfo` and `JsonSerializerOptions` once at startup for each endpoint.

https://github.com/dotnet/aspnetcore/blob/main/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointJsonResponseEmitter.cs#L14-L15

I think it's fine to take only `JsonSerializerOptions` for this API.

Reply:
> I think it's fine to take only JsonSerializerOptions for this API.

Did you mean `JsonTypeInfo<T>` instead? If we just took JsonSerializerOptions, it wouldn't make sense to pre-compute the JsonTypeInfo. Because we would need to look it up again if we just had the options.

Reply:
I've removed the `JsonSerializerOptions` from here and in RDF.